### PR TITLE
Update to PROJ 8.1.0

### PR DESCRIPTION
**Pull request by urschrei**
- Update proj source
- Update proj-sys
- Update generated bindings
- Update dependencies

Merging is blocked on https://github.com/georust/docker-images/pull/13

### CODE REVIEW

**Review by lnicola — APPROVED** [review]

**Comment by michaelkirk** [comment]
bors try

**Comment by urschrei** [comment]
bors try

**Comment by urschrei** [comment]
Tests were never gonna pass on the outdated images, were they?

**Comment by urschrei** [comment]
bors try

**Comment by urschrei** [comment]
Now we're getting somewhere (why are those tests failing on ubuntu?)

**Comment by michaelkirk** [comment]
> Now we're getting somewhere (why are those tests failing on ubuntu?)

FYI, I've just reproduced this failure locally:

```
$ docker run -v /Users/mkirk/src/georust/docker-images/rust-1.51/../..:/tmp/georust -w /tmp/georust/proj docker.io/georust/proj-ci:rust-1.51 cargo test --no-default-features

failures:                                                                                                                                                                     
                                                                                                                                                                              
---- proj::test::test_conversion stdout ----                                                                                                                                  
thread 'proj::test::test_conversion' panicked at 'assert_relative_eq!(t.x(), 1450880.2910605017)                                                                              
                                                                                                                                                                              
    left  = 1450880.2910605012                                                                                                                                                
    right = 1450880.2910605017                                                                                                                                                
                                                                                                                                                                              
', src/proj.rs:1103:9                                                                                                                                                         
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- proj::test::test_inverse_projection stdout ----
thread 'proj::test::test_inverse_projection' panicked at 'assert_relative_eq!(t.x(), 0.43633200013698786)

    left  = NaN
    right = 0.43633200013698786

', src/proj.rs:1066:9

---- proj::test::test_london_inverse stdout ----
thread 'proj::test::test_london_inverse' panicked at 'assert_relative_eq!(t.x(), 0.0023755864830313977)

    left  = NaN
    right = 0.0023755864830313977

', src/proj.rs:1083:9

---- proj::test::test_projection stdout ----
thread 'proj::test::test_projection' panicked at 'assert_relative_eq!(t.x(), 500119.7035366755, epsilon = 1e-5)

    left  = NaN
    right = 500119.7035366755

', src/proj.rs:1051:9

failures:
    proj::test::test_conversion
    proj::test::test_inverse_projection
    proj::test::test_london_inverse
    proj::test::test_projection

test result: FAILED. 11 passed; 4 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.48s

error: test failed, to rerun pass '--lib'
```

**Comment by michaelkirk** [comment]
... and interestingly with proj-ci:1.53, I get only 1 of the same 4 failures as with proj-ci:1.51...

```
$ docker run -v /Users/mkirk/src/georust/docker-images/rust-1.53/../..:/tmp/georust -w /tmp/georust/proj docker.io/georust/proj-ci:rust-1.53 cargo test --no-default-features
---- proj::test::test_conversion stdout ----
thread 'proj::test::test_conversion' panicked at 'assert_relative_eq!(t.x(), 1450880.2910605017)

    left  = 1450880.2910605012
    right = 1450880.2910605017

', src/proj.rs:1103:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    proj::test::test_conversion

test result: FAILED. 14 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.46s

error: test failed, to rerun pass '--lib'
```

**Comment by urschrei** [comment]
bors try

**Comment by urschrei** [comment]
I have no idea why we were seeing `NaN` errors on 1.51, and that is somewhat concerning, but:
I've fixed the "actual" error we were seeing by using a known-good string for the NAD83 (feet) to NAD83 (meters) conversion by feeding PROJ the EPSG codes, then generating the pipeline string. I don't know where the original string came from (mea culpa), but the source of the replacement is now noted inline.

File: src/proj.rs [hunk 1]
```diff
@@ -1186,7 +1186,7 @@ mod test {
         assert_eq!(area.west, -35.58);
         assert_eq!(area.south, 24.6);
         assert_eq!(area.east, 44.83);
-        assert_eq!(area.north, 84.17);
```
Comment (michaelkirk) [inline src/proj.rs:1189]: This is a pretty large change - curious!

**Review by michaelkirk — APPROVED** [review]
> Test failures due to different values when we update to a newer PROJ version aren't concerning (to me) if they're small (the seventh decimal place is 1.11 cm at the equator) since the values used in the EPSG db used by PROJ are updated periodically, so long as we're using known-to-be-correct inputs – the alternative is playing epistemological whack-a-mole with at least two unknowns, and nobody's got time for that.

This sounds reasonable to me. Thanks for fixing CI!

File: src/proj.rs (hunk 1 above)
Comment (urschrei) [reply src/proj.rs:1189]: It certainly is, but this isn't even a value we're calculating. Perhaps I'll ask on the list…

Comment (urschrei) [reply src/proj.rs:1189]: Extent changed January 2021: https://epsg.org/closed-change-request/browse/id/2020.127
New extent, matching what we now get from the DB: https://epsg.org/extent_2881/Europe-LCC-LAEA.html
CRS: https://epsg.org/crs_3035/ETRS89-extended-LAEA-Europe.html

Comment (michaelkirk) [reply src/proj.rs:1189]: The fast moving world of geodesy!

Fascinating.

**Comment by urschrei** [comment]
bors r=michaelkirk

**Pull request merged**